scripts/stac/imagery/collection.py

```diff
@@ -75,6 +84,50 @@ def __init__(
 
         self.add_providers(providers)
 
+    def add_capture_area(self, polygons: List[shapely.geometry.shape], target: str) -> None:
+        """Add the capture area of the Collection.
+        The `href` or path of the capture-area.geojson is always set as the relative `./capture-area.geojson`
+
+        Args:
+            polygons: list of geometries
+            target: path of the capture-area-geojson file
+        """
+
+        # The GSD is currently alway in meters (e.g. `0.3m`)
+        capture_area_content = generate_capture_area(polygons, float(self.metadata["gsd"].replace("m", "")))
+        with TemporaryDirectory() as tmp_path:
+            tmp_capture_area_path = os.path.join(tmp_path, CAPTURE_AREA_FILE_NAME)
+            write(
+                tmp_capture_area_path,
+                json.dumps(capture_area_content).encode("utf-8"),
+                content_type=ContentType.GEOJSON.value,
+            )
+
+            file_stats = os.stat(tmp_capture_area_path)
+            file_checksum = checksum.multihash_as_hex(tmp_capture_area_path)
+            capture_area = {
+                "href": f"./{CAPTURE_AREA_FILE_NAME}",
+                "title": "Capture area",
+                "type": ContentType.GEOJSON,
+                "roles": ["metadata"],
+                "file:checksum": file_checksum,
+                "file:size": file_stats.st_size,
+            }
+            self.stac.setdefault("assets", {})["capture_area"] = capture_area
+
+            # Save `capture-area.geojson` in target
+            write(
+                os.path.join(target, CAPTURE_AREA_FILE_NAME),
+                read(tmp_capture_area_path),
+                content_type=ContentType.GEOJSON.value,
+            )
+
+        if not self.stac.get("stac_extensions"):
+            self.stac["stac_extensions"] = []
```

Comment:
You can make the conditional implicit by doing this:

```suggestion
        self.stac["stac_extensions"] = self.stac.get("stac_extensions", [])
```